src/Tools/dotnet-monitor/Egress/AzureBlob/AzureBlobEgressProvider.cs

```diff
@@ -52,6 +55,12 @@ public AzureBlobEgressProvider(ILogger<AzureBlobEgressProvider> logger)
 
                 string blobUriString = GetBlobUri(blobClient);
                 Logger?.EgressProviderSavedStream(EgressProviderTypes.AzureBlobStorage, blobUriString);
+
+                if (CheckQueueEgressOptions(options))
+                {
+                    await EgressMessageToQueue(artifactSettings.Name, options, token);
```

Comment:
It cannot just be the artifact name that is passed in. It most likely needs to be the reuslt of the `GetBlobName` call from above because the name of the blob can contain "folders".

For example, for the URI `https://dotnetcli.azureedge.net/dotnet/diagnostics/monitor/6.1.0/dotnet-monitor.6.1.0.nupkg`:
- The account is `dotnetcli`
- The container is `dotnet`
- The blob is `diagnostics/monitor/6.1.0/dotnet-monitor.6.1.0.nupkg`

Setting just `dotnet-monitor.6.1.0.nupkg` as the queue item would not be sufficient for the association within an Azure function to automatically create a stream from that since it is missing `diagnostics/monitor/6.1.0`.

Reply:
I changed this in the latest version; in my example I actually was using a blob that was more than just the artifact name (artifacts/artifact_name), but I had just changed the Azure Function so that it knew to look in the artifacts directory so this would work. Removing the need for Azure Function users to do that step should make it easier, however.